.github/workflows/weekly.yml

```diff
@@ -0,0 +1,197 @@
+name: weekly tests
+
+on:
+  schedule:
+  - cron: 37 7 * * 1
+  workflow_dispatch:
+  pull_request:
+    types: [labeled, opened, synchronize, reopened]
+    branches:
+    - main
+
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+jobs:
+
+  weekly:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    if: >-
+      github.event_name == 'workflow_dispatch' ||
+      (github.event_name == 'pull_request' && contains(github.event.pull_request.labels.*.name, 'run weekly tests in CI')) ||
+      (github.event_name == 'schedule' && github.repository == 'PlasmaPy/PlasmaPy' && github.ref == 'refs/heads/main')
```

Comment:
📝 This is saying that we should run tests...

 - If we manually trigger it (via workflow dispatch)
 - It's a pull request that has been labeled with "run weekly tests in CI"
 - As a cron job if it's on the `main` branch of `PlasmaPy/PlasmaPy`.